### ISSUE TITLE
MBP M1 native homebrew packages doesn't result anything without sudo

### ISSUE THREAD

**Issue by hjoshi123** [Homebrew/homebrew-core#72225]
**`brew config` output**

HOMEBREW_VERSION: 3.0.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: b712bcc01e8fdaae25d6f232f51637bf00db307f
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 97e9d1f96b976d74aa0736fac65aa7ad890035b9
Core tap last commit: 4 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 12.0 build 1200
Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.64.1 => /usr/bin/curl
macOS: 11.2.2-arm64
CLT: 12.4.0.0.1.1610135815
Xcode: N/A
Rosetta 2: false

**`brew doctor` output**

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

****

- [X] The `brew doctor` above contains no "Warning" lines.

**What were you trying to do (and why)?**

Was trying to run `go get github.com/gin-gonic/gin` but isn't running. When done the same command with sudo i.e. `sudo go get github.com/gin-gonic/gin` it works (the module gets installed). The same problem occurs when I open VSCode as it is also not able to resolve the GOPATH inside the `opt/homebrew`  but is working when I run the VSCode as root

**What happened (include all command output)?**

`go: cannot find GOROOT directory: /opt/homebrew/bin/go/libexec` This is the output I get without the sudo. I have checked the ownership of /opt/homebrew its mine i.e. `whoami`. So I am not sure what is causing this problem.

**What did you expect to happen?**

To Install the module successfully and for VSCode to resolve the GOROOT and other dependencies.

**Step-by-step reproduction instructions (by running `brew` commands)**

After installing the M1 setup in `/opt/homebrew` try running the following
`brew install go`
`export PATH=/opt/homebrew/bin:$PATH`
`go get github.com/gin-gonic/gin`

**Comment by SMillerDev** [comment]
What about `/opt/homebrew/bin/go/libexec`? Does it exist? Who is the owner?

**Comment by hjoshi123** [comment]
yes it does exist and the owner is `hemantjoshi` i.e `whoami`. Attaching the `ls -aril` command output of `/opt/homebrew`

```sh
3853261 drwxr-xr-x   5 hemantjoshi  staff   160 Feb 24 23:42 var
4833804 drwxr-xr-x   8 hemantjoshi  staff   256 Feb 24 23:42 share
4833802 drwxr-xr-x   3 hemantjoshi  staff    96 Feb 20 19:58 sbin
4833800 drwxr-xr-x   4 hemantjoshi  staff   128 Feb 28 12:42 opt
3851222 drwxrwxr-x   4 hemantjoshi  wheel   128 Feb 24 23:42 manpages
4833798 drwxr-xr-x   3 hemantjoshi  staff    96 Feb 20 19:58 lib
4833796 drwxr-xr-x   3 hemantjoshi  staff    96 Feb 20 19:58 include
4833794 drwxr-xr-x   7 hemantjoshi  staff   224 Feb 24 23:42 etc
3851145 drwxrwxr-x  63 hemantjoshi  wheel  2016 Feb 24 23:42 docs
3851136 drwxrwxr-x   7 hemantjoshi  wheel   224 Feb 20 19:57 completions
3851134 drwxrwxr-x   7 hemantjoshi  wheel   224 Feb 28 12:42 bin
```

**Comment by gromgit** [comment]
> > What about `/opt/homebrew/bin/go/libexec`? Does it exist? Who is the owner?

> yes it does exist and the owner is `hemantjoshi` i.e `whoami`

Then something has gone VERY wrong. `/opt/homebrew/bin/go/libexec` cannot exist, since `/opt/homebrew/bin/go` should be a symlink that points to a file. Please double-check the output of `ls -l /opt/homebrew/bin/go/libexec`, as you should get a "file not found" error or something similar.

More importantly, what's the output of `echo $GOROOT`? If it's `/opt/homebrew/bin/go/libexec`, then you set it wrong...and you shouldn't have to set it at all. `unset GOROOT` and try running your Go command again.

**Comment by hjoshi123** [comment]
Ah!! Ok doing the `unset GOROOT` worked. I would love to know the reason why it worked though. When I setup Go in the x86_64 Mac I did the `GOROOT` which was `/usr/local/bin/go/libexec` (not sure don't remember it properly) but it was working fine. With M1 what is happening or is it something with Go that I am not aware of? Thank you for the help and if you could tell me why it worked it would be helpful.

**Comment by carlocab** [comment]
`/usr/local/bin/go/libexec` is a path that shouldn't exist, so it's a bit surprising that didn't break things before. (If the path did exist, then your go install must've been very broken.)

Can we close this issue now?

**Comment by hjoshi123** [comment]
Oh ok. So ideally there should be no `GOROOT` path set? 
Yeah thanks for the help sure we can close this issue?

**Comment by carlocab** [comment]
Yes, I think you only need to set `GOROOT` if Go isn't working without it. I actually think you shouldn't need to set this at all with newer versions of Go.
